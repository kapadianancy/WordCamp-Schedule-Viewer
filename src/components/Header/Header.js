import React from 'react';
import NancyProfile from '../../../public/assets/images/nancy-profile.jpeg';

function Header(props) {
    var divStyle = {
        color: 'red'
    }
    var style={
        height:70
    }
       
    return (
        <>
            <div className="navbar navbar-expand-md navbar-dark" style={style}>
                <div className="navbar-brand">
                    <a href="#" className="d-inline-block">
                        {/* <img src="../assets/images/logo_light.png" alt=""/> */}
                        <h1 style={{color: "white", fontWeight: "bold"}}>WordCamp Schedule Viewer</h1>
                    </a>
                </div>


                <div className="collapse navbar-collapse" id="navbar-mobile">
                    <ul className="navbar-nav">
                        <li className="nav-item">
                            <a href="#"
                               className="navbar-nav-link sidebar-control sidebar-main-toggle d-none d-md-block">
                                <i className="icon-paragraph-justify3"></i>
                            </a>
                        </li>


                    </ul>

                    <span  id="today-date" className="ml-md-3 mr-md-auto" style={{fontSize:"20px"}}>Today's Date</span>

                    <ul className="navbar-nav">


                        <li className="nav-item dropdown dropdown-user">
                            <a href="#" className="navbar-nav-link d-flex align-items-center dropdown-toggle"
                               data-toggle="dropdown">
                                 <img src={NancyProfile} className="rounded-circle mr-2" height="34" alt=""/>
                                {/*<Avatar name={user == null ? null : user.name} round={true} size="50" color="#26a69a"*/}
                                {/*        style={{margin: "5px 10px 10px auto"}}/>*/}
                                <span>Nancy Kapadia</span>
                            </a>

                            <div className="dropdown-menu dropdown-menu-right">

                            </div>
                        </li>
                    </ul>
                </div>
            </div>
        </>
    );
}

export default Header;