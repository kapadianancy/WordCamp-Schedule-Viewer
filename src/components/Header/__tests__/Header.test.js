/* eslint-disable no-undef */
/**
 * @jest-environment  jsdom
 */

import React from 'react'
import { cleanup, render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import Header from '../Header'

describe('#Header', () => {
  const props = {
    themeClass: 'dark',
    themes: ['full-moon'],
    changeTheme: jest.fn()
  }

  afterEach(() => {
    cleanup()
  })

  it('should render successfully', () => {
    const { getByTestId } = render(<Header></Header>)
    const component = getByTestId('header-test')
    expect(component).toBeInTheDocument()
  })

  it('should render theme and height correctly', () => {
    const { getByTestId } = render(<Header {...props}></Header>)
    const component = getByTestId('header-test')
    expect(component.classList.contains('dark')).toBe(true)
    expect(component.style.height).toBe('70px')
  })

  it('should render title correctly', () => {
    const { getByTestId } = render(<Header {...props}></Header>)
    const component = getByTestId('header-text-test')
    expect(component).toHaveTextContent('W-ScheduleViewer')
    expect(component.classList.contains('dark')).toBe(true)
    expect(component.style.fontWeight).toBe('bold')
  })

  it("should contain today's date correctly", () => {
    const { getByTestId } = render(<Header></Header>)
    const component = getByTestId('today-test')
    expect(component).toHaveTextContent(new Date().toDateString())
  })

  it('should render profile img correctly', () => {
    const { getByTestId } = render(<Header></Header>)
    const component = getByTestId('profile-img-test')
    expect(component).toBeInTheDocument()
    expect(component).toHaveAttribute('height', '34')
    expect(component.getAttribute('src')).not.toBe(null)
  })

  it('should change theme to:full-moon', () => {
    const component = render(<Header {...props}></Header>)
    const testID = component.getByTestId('full-moon-test')
    fireEvent.click(testID)
    expect(props.changeTheme).toHaveBeenCalledTimes(1)
  })

  it('should match the snapshot', () => {
    const tree = renderer.create(<Header {...props}></Header>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
