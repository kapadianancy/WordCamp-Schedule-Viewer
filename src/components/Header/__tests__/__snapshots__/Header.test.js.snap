// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Header should match the snapshot 1`] = `
<div
  className="navbar navbar-expand-md dark"
  data-testid="header-test"
  style={
    Object {
      "height": 70,
    }
  }
>
  <div
    className="navbar-brand"
  >
    <a
      className="d-inline-block"
      href="#"
    >
      <h1
        className="dark"
        data-testid="header-text-test"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        W-ScheduleViewer
      </h1>
    </a>
  </div>
  <div
    className="collapse navbar-collapse"
    id="navbar-mobile"
  >
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item"
      >
        <a
          className="navbar-nav-link sidebar-control sidebar-main-toggle d-none d-md-block"
          href="#"
        >
          <i
            className="icon-paragraph-justify3"
          />
        </a>
      </li>
    </ul>
    <span
      className="ml-md-3 mr-md-auto"
      data-testid="today-test"
      id="today-date"
      style={
        Object {
          "fontSize": "20px",
        }
      }
    >
      Sat Jul 17 2021
    </span>
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item dropdown dropdown-user"
      >
        <a
          className="navbar-nav-link d-flex align-items-center dropdown-toggle dark"
          data-toggle="dropdown"
        >
          <img
            alt=""
            className="rounded-circle mr-2"
            data-testid="profile-img-test"
            height="34"
            src={
              Object {
                "process": [Function],
              }
            }
          />
          <span>
            Nancy Kapadia
          </span>
        </a>
        <div
          className="dropdown-menu dropdown-menu-right"
        >
          <a
            className="dropdown-item"
            data-testid="full-moon-test"
            onClick={[Function]}
          >
            <img
              src={
                Object {
                  "process": [Function],
                }
              }
            />
            Full Moon
          </a>
          <a
            className="dropdown-item"
            data-testid="eclipse-test"
            onClick={[Function]}
          >
            <img
              src={
                Object {
                  "process": [Function],
                }
              }
            />
             Eclipse
          </a>
          <a
            className="dropdown-item"
            data-testid="new-moon-test"
            onClick={[Function]}
          >
            <img
              src={
                Object {
                  "process": [Function],
                }
              }
            />
             New Moon
          </a>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
